wordpress-mobile/WordPress-iOS
# Move stats code into WPiOS as a dynamic framework

Issue #7120 

This PR does a bunch of things and shouldn't require a ton of testing but should be reviewed carefully for an understanding of what's happened to the project.

* A new project was created called WordPressComStatsiOS. This project has two targets - a dynamic framework and a test target. All of the code and configuration settings were made to mirror what the original stats project podspec did. I chose to not make a resource bundle like I did in the podspec so a few things had to change in WPiOS to load the SiteStats Storyboard properly.
* Stats requires CocoaPods because of things like networking, logging, etc. The Podfile was updated to support this new project and shared pod definitions were placed into a macro of sorts to make version bumps easier. The stats project is in it's own root-level target which also means the dependency of the project on the main WordPress target is unknown to CocoaPods.
* Manually configured the main target to embed the framework. Also added the framework to the linker for the today widget.
* Added the stats unit tests to the scheme for unit tests in the main target.

## To Test
1. Update your local CocoaPods repo with `pod repo update`.
2. Optionally clean your `DerivedData` folder.
3. Build & test main WordPress target. All tests should pass - UI tests are turned off now with this PR.
4. Run main WordPress target.
5. View stats for a site. Make sure nothing crashes, icons load on screens.
6. Try viewing stats via a stats notification. Make sure nothing crashes.
7. Try viewing stats on a published post by tapping the stats button in a post list.
8. Stop the running app.
9. Switch to the WordPressComStatsiOS scheme and run unit tests.

See if you can break it any other way.

Needs review: @elibud, @nheagy, @koke, @frosty, @kurzee
